quinn-udp/src/unix.rs

```diff
@@ -794,6 +794,56 @@ pub(crate) const BATCH_SIZE: usize = 32;
 #[cfg(apple_slow)]
 pub(crate) const BATCH_SIZE: usize = 1;
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod linux {
+    // https://www.linfo.org/kernel_version_numbering.html
+    pub(crate) struct KernelVersion {
+        version: u8,
+        major_revision: u8,
+        pub(crate) string: String,
+    }
+
+    impl KernelVersion {
+        pub(crate) fn from_uname() -> Option<Self> {
+            let mut n = unsafe { std::mem::zeroed() };
+            let r = unsafe { libc::uname(&mut n) };
+            if r != 0 {
+                crate::log::trace!("Failed to run libc::uname ({})", r);
+                return None;
```

Comment:
You can create `std::io::Error::other` here and for all the other failure cases too. Then, you only need to log in a single place as to why GSO is not supported:

- kernel version too low
- some IO / parsing error in the check

I'd almost say the IO error should be a WARN or an ERROR?

Reply:
> Then, you only need to log in a single place as to why GSO is not supported:

If it's only for this reason, I could also return `Result<Self, String>`. Or do you prefer `Result<Self, std::io::Error>`?

Reply:
No strong preference.